sdk/storage/azblob/lease/blob_client.go

```diff
@@ -0,0 +1,141 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package lease
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/appendblob"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/blob"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/blockblob"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/internal/base"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/internal/generated"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/internal/shared"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/pageblob"
+)
+
+// BlobClient provides lease functionality for the underlying blob client.
+type BlobClient struct {
+	blobClient *blob.Client
+	leaseID    *string
+}
+
+// BlobClientOptions contains the optional values when creating a BlobClient.
+type BlobClientOptions struct {
+	// LeaseID contains a caller-provided lease ID.
+	LeaseID *string
+}
+
+// NewBlobClient creates a blob lease client for the provided blob client.
+func NewBlobClient[T appendblob.Client | blob.Client | blockblob.Client | pageblob.Client](client *T, options *BlobClientOptions) (*BlobClient, error) {
```

Comment:
The approach used here is the same as C#.  The lease client is built from an existing client.